tests/test_minhash.py

```diff
@@ -1196,6 +1196,20 @@ def test_set_abundance_clear_4():
     a.set_abundances({20: 1, 10: 2}, clear=False)
     assert a.hashes == {10: 3, 20: 3}
 
+def test_clear_abundance_on_zero():
+    mh = sourmash.minhash.MinHash(n=0, ksize=31, scaled=1, track_abundance=True)
+    mh.set_abundances({ 1: 5, 2: 3, 3 : 5 })
+    mh.set_abundances({ 1: 0 }, clear=False)
+    assert 1 not in dict(mh.hashes)
+    assert dict(mh.hashes)[2] == 3
+    assert dict(mh.hashes)[3] == 5
+    assert len(mh) == 2
+
+    with pytest.raises(ValueError):
+        mh.set_abundances({ 2: -1 }) # Test on clear = True
+        mh.set_abundances({ 2: -1 }, clear=False)
```

Comment:
Should this be two separate `with` blocks? It is more code, but it is clearer that both statements should raise `ValueError` (if only one raises the exception the test still passes)